### PR TITLE
Feature/stringlike value indexing

### PR DESCRIPTION
**Pull request by slorello89**
New feature to allow for indexing & querying by enums, ulid's, GUIDs, booleans per #128

### CODE REVIEW

**Comment by VagyokC4** [comment]
@slorello89 Also, after this feature is approved and merged, how long before we can get a fresh NuGet release?

File: src/Redis.OM/Modeling/TypeDeterminationUtilities.cs [hunk 1]
```diff
@@ -64,5 +67,14 @@ internal static SearchFieldType GetSearchFieldType(Type type)
 
             throw new ArgumentException("Unrecognized Index type, can only index numerics, GeoLoc, or String");
```
Comment (shacharPash) [inline src/Redis.OM/Modeling/TypeDeterminationUtilities.cs:68]: If I'm not missing something here, you need to update this line with the other types (Ulid, Enum, bool..)

File: src/Redis.OM/Common/ExpressionTranslator.cs [hunk 2]
```diff
@@ -561,6 +576,30 @@ private static string TranslateBinaryExpression(BinaryExpression binExpression)
...
+                    if (attr != null && attr.ConverterType == typeof(JsonStringEnumConverter))
+                    {
+                        if (int.TryParse(rightContent, out int ordinal))
+                        {
+                            rightContent = Enum.ToObject(member.Type, ordinal).ToString();
+                        }
+                    }
+                    else
+                    {
+                        if (!int.TryParse(rightContent, out _))
```
Comment (shacharPash) [inline src/Redis.OM/Common/ExpressionTranslator.cs:592]: this is the only "if", why didn't you write just "else if"?

File: src/Redis.OM/Common/ExpressionTranslator.cs [hunk 3]
```diff
@@ -561,6 +576,30 @@ private static string TranslateBinaryExpression(BinaryExpression binExpression)
                     sb.Append(predicate);
                     sb.Append(")");
                 }
+                else if (binExpression.Left is UnaryExpression uni)
+                {
+                    member = (MemberExpression)uni.Operand;
+                    var attr = member.Member.GetCustomAttributes(typeof(JsonConverterAttribute)).FirstOrDefault() as JsonConverterAttribute;
+                    if (attr != null && attr.ConverterType == typeof(JsonStringEnumConverter))
+                    {
+                        if (int.TryParse(rightContent, out int ordinal))
```
Comment (shacharPash) [inline src/Redis.OM/Common/ExpressionTranslator.cs:585]: Why didn't you add this condition to the if above with "&&"?

File: src/Redis.OM/Common/ExpressionTranslator.cs (hunk 2 above)
Comment (slorello89) [reply src/Redis.OM/Common/ExpressionTranslator.cs:592]: We want to check whether or not it has the attribute, and then we want to check if the content is already set up correctly. If you include the extra conditional in the surrounding if/else the truth-tables don't quite align. Applies to both of these comments.

File: test/Redis.OM.Unit.Tests/RediSearchTests/SearchFunctionalTests.cs [hunk 4]
```diff
@@ -504,5 +505,91 @@ public async Task FindByKeyAsync()
...
+            var anEnum = AnEnum.two;
+            var three = AnEnum.three;
+            var collection = new RedisCollection<ObjectWithStringLikeValueTypes>(_connection);
+            var result = await collection.Where(x => x.AnEnum == AnEnum.two).ToListAsync();
+            Assert.NotEmpty(result);
+            result = await collection.Where(x => x.AnEnum == anEnum).ToListAsync();
+            Assert.NotEmpty(result);
+            result = await collection.Where(x => x.AnEnum == obj.AnEnum).ToListAsync();
+            Assert.NotEmpty(result);
+            result = await collection.Where(x => (int)x.AnEnumAsInt > 1).ToListAsync();
```
[the remark this answers is not in the document]
Comment (slorello89) [reply test/Redis.OM.Unit.Tests/RediSearchTests/SearchFunctionalTests.cs:567]: I added an extra test to check this condition. Seems like it works just fine.

File: src/Redis.OM/Modeling/TypeDeterminationUtilities.cs (hunk 1 above)
Comment (slorello89) [reply src/Redis.OM/Modeling/TypeDeterminationUtilities.cs]: Updated

**Comment by slorello89** [comment]
> @slorello89 Also, after this feature is approved and merged, how long before we can get a fresh NuGet release?

After this is merged I'm going to bump the version to 0.2.0 and cut a relase.

**Review by shacharPash — APPROVED** [review]

**Pull request merged**